indykite/resource_entity_matching_pipeline.go

```diff
@@ -102,14 +127,28 @@ func resourceEntityMatchingPipelineBuild(
 	_ *ClientContext,
 	builder *config.NodeRequest,
 ) {
-	sourceNoteTypes := rawArrayToTypedArray[string](data.Get(entityMatchingPipelineSourceNodeFilterKey))
-	targetNodeTypes := rawArrayToTypedArray[string](data.Get(entityMatchingPipelineTargetNodeFilterKey))
+	cfg := &configpb.EntityMatchingPipelineConfig{}
 
-	cfg := &configpb.EntityMatchingPipelineConfig{
-		NodeFilter: &configpb.EntityMatchingPipelineConfig_NodeFilter{
-			SourceNodeTypes: sourceNoteTypes,
-			TargetNodeTypes: targetNodeTypes,
-		},
+	// Check if SourceNodeFilterKey and TargetNodeFilterKey are available
+	sourceNodeTypes, hasSourceKey := data.GetOk(entityMatchingPipelineSourceNodeFilterKey)
+	targetNodeTypes, hasTargetKey := data.GetOk(entityMatchingPipelineTargetNodeFilterKey)
+
+	if hasSourceKey && hasTargetKey {
+		// Convert sourceNodeTypes and targetNodeTypes to typed array if they exist
+		cfg.NodeFilter = &configpb.EntityMatchingPipelineConfig_NodeFilter{
+			SourceNodeTypes: rawArrayToTypedArray[string](sourceNodeTypes),
+			TargetNodeTypes: rawArrayToTypedArray[string](targetNodeTypes),
+		}
+	}
+	// Check if SimilarityScoreCutOffKey is available
+	if similarityScoreCutoff, hasSimilarityScore := data.GetOk(
+		entityMatchingPipelineSimilarityScoreCutOffKey); hasSimilarityScore {
+		// Convert similarityScoreCutoff to float32 if it exists
+		if similarityScoreCutoff, ok := similarityScoreCutoff.(float64); ok {
+			cfg.SimilarityScoreCutoff = float32(similarityScoreCutoff)
+		}
 	}
```

Comment:
1. You don't need to check casting to `float64`. It will be always true because you defined that in schema.

Reply:
updated